### ISSUE TITLE
[Bug]: [910B][Triton] GDN prefill kernel intermittently leaves output elements unwritten on first launches — first inference returns all-NaN; root cause needs investigation

### ISSUE THREAD

**Issue by Absoler** [vllm-project/vllm-ascend#17009]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
PyTorch version: 2.10.0+cpu
Is debug build: False

OS: Ubuntu 22.04.5 LTS (aarch64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version: 15.0.7
CMake version: version 4.4.2
Libc version: glibc-2.35

Python version: 3.12.13 (main, Aug  3 2026, 03:44:26) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-91-generic-aarch64-with-glibc2.35

CPU:
Architecture:                       aarch64
CPU op-mode(s):                     64-bit
Byte Order:                         Little Endian
CPU(s):                             192
On-line CPU(s) list:                0-191
Vendor ID:                          HiSilicon
Model name:                         Kunpeng-920
Model:                              0
Thread(s) per core:                 1
Core(s) per socket:                 48
Socket(s):                          4
Stepping:                           0x1
BogoMIPS:                           200.00
Flags:                              fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma dcpop asimddp asimdfhm ssbs
L1d cache:                          12 MiB (192 instances)
L1i cache:                          12 MiB (192 instances)
L2 cache:                           96 MiB (192 instances)
L3 cache:                           192 MiB (8 instances)
NUMA node(s):                       8
NUMA node0 CPU(s):                  0-23
NUMA node1 CPU(s):                  24-47
NUMA node2 CPU(s):                  48-71
NUMA node3 CPU(s):                  72-95
NUMA node4 CPU(s):                  96-119
NUMA node5 CPU(s):                  120-143
NUMA node6 CPU(s):                  144-167
NUMA node7 CPU(s):                  168-191
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; __user pointer sanitization
Vulnerability Spectre v2:           Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] pyzmq==27.1.0
[pip3] torch==2.10.0+cpu
[pip3] torch_npu==2.10.0.post4
[pip3] torchaudio==2.10.0+cpu
[pip3] torchvision==0.25.0+cpu
[pip3] transformers==5.5.4
[conda] Could not collect
vLLM Version: 0.23.0
vLLM Ascend Version: 0.23.0

ENV Variables:
ASCEND_TOOLKIT_LATEST_HOME=/usr/local/Ascend/ascend-toolkit/latest
ATB_OPSRUNNER_KERNEL_CACHE_LOCAL_COUNT=1
ATB_STREAM_SYNC_EVERY_RUNNER_ENABLE=0
ATB_STREAM_SYNC_EVERY_KERNEL_ENABLE=0
ATB_CXX_ABI=1
ATB_OPSRUNNER_KERNEL_CACHE_GLOABL_COUNT=5
ATB_HOME_PATH=/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1
ASCEND_TOOLKIT_HOME=/usr/local/Ascend/cann-9.1.0
ATB_COMPARE_TILING_EVERY_KERNEL=0
ASCEND_OPP_PATH=/usr/local/Ascend/cann-9.1.0/opp
LD_LIBRARY_PATH=/usr/local/Ascend/cann-9.1.0/lib64:/usr/local/Ascend/cann-9.1.0/lib64/plugin/opskernel:/usr/local/Ascend/cann-9.1.0/lib64/plugin/nnengine:/usr/local/Ascend/cann-9.1.0/opp/built-in/op_impl/ai_core/tbe/op_tiling:/usr/local/Ascend/driver/lib64:/usr/local/Ascend/driver/lib64/common:/usr/local/Ascend/driver/lib64/driver:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/lib:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/examples:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/tests/atbopstest:/usr/local/Ascend/ascend-toolkit/latest/tools/aml/lib64:/usr/local/Ascend/ascend-toolkit/latest/tools/aml/lib64/plugin:/usr/local/Ascend/ascend-toolkit/latest/lib64:/usr/local/Ascend/ascend-toolkit/latest/lib64/plugin/opskernel:/usr/local/Ascend/ascend-toolkit/latest/lib64/plugin/nnengine:/usr/local/Ascend/ascend-toolkit/latest/opp/built-in/op_impl/ai_core/tbe/op_tiling:/usr/local/python3.12.13/lib::/usr/local/lib
ASCEND_AICPU_PATH=/usr/local/Ascend/cann-9.1.0
ATB_STREAM_SYNC_EVERY_OPERATION_ENABLE=0
ASCEND_HOME_PATH=/usr/local/Ascend/cann-9.1.0
ATB_MATMUL_SHUFFLE_K_ENABLE=1
ATB_WORKSPACE_MEM_ALLOC_ALG_TYPE=1
ATB_SHARE_MEMORY_NAME_SUFFIX=
TORCH_DEVICE_BACKEND_AUTOLOAD=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_root
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1


NPU:
+------------------------------------------------------------------------------------------------+
| npu-smi 25.5.1                   Version: 25.5.1                                               |
+---------------------------+---------------+----------------------------------------------------+
| NPU   Name                | Health        | Power(W)    Temp(C)           Hugepages-Usage(page)|
| Chip                      | Bus-Id        | AICore(%)   Memory-Usage(MB)  HBM-Usage(MB)        |
+===========================+===============+====================================================+
| 0     910B4               | OK            | 87.2        39                0    / 0             |
| 0                         | 0000:C1:00.0  | 0           0    / 0          2875 / 32768         |
+===========================+===============+====================================================+
| 1     910B4               | OK            | 97.1        40                0    / 0             |
| 0                         | 0000:01:00.0  | 0           0    / 0          2875 / 32768         |
+===========================+===============+====================================================+
| 2     910B4               | OK            | 89.8        38                0    / 0             |
| 0                         | 0000:C2:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
| 3     910B4               | OK            | 93.9        39                0    / 0             |
| 0                         | 0000:02:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
| 4     910B4               | OK            | 87.6        39                0    / 0             |
| 0                         | 0000:81:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
| 5     910B4               | OK            | 90.5        39                0    / 0             |
| 0                         | 0000:41:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
| 6     910B4               | OK            | 94.0        49                0    / 0             |
| 0                         | 0000:82:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
| 7     910B4               | OK            | 87.6        37                0    / 0             |
| 0                         | 0000:42:00.0  | 0           0    / 0          2858 / 32768         |
+===========================+===============+====================================================+
+---------------------------+---------------+----------------------------------------------------+
| NPU     Chip              | Process id    | Process name             | Process memory(MB)      |
+===========================+===============+====================================================+
| No running processes found in NPU 0                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 1                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 2                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 3                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 4                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 5                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 6                                                            |
+===========================+===============+====================================================+
| No running processes found in NPU 7                                                            |
+===========================+===============+====================================================+

CANN:
package_name=Ascend-cann-toolkit
version=9.1.0
innerversion=V100R001C11SPC001B243
compatible_version=[V100R001C15],[V100R001C21],[V100R001C23],[V100R001C10]
arch=aarch64
os=linux
path=/usr/local/Ascend/cann-9.1.0
```

</details>

### 🐛 Describe the bug

derived from https://github.com/vllm-project/vllm-ascend/issues/15560

On v0.23.0, serving Qwen3.6-27B (qwen3_5 arch, GDN linear-attention layers),
**the first inference after engine startup returns garbage**: all-NaN logits,
greedy decoding collapses to token id 0 (repeated `!`). Every subsequent
request is normal, which makes it look like a warmup issue — it is not.

We bisected the corruption with stage-level NaN/Inf tracing inside the GDN
chunk pipeline and found that **`recompute_w_u_fwd_kernel`
(`vllm_ascend/ops/triton/fla/wy_fast.py`) leaves a few valid elements of its
`u` / `w` outputs unwritten on the process's first real prefill**. With
`torch.empty_like()`-allocated outputs, those positions keep uninitialized
memory; when it contains NaN, a single element is enough to poison the whole
model (u → AscendC fwd_h → fwd_o → out_proj matmul → all-NaN hidden states).

The zero-initialization of the outputs in the linked PR only **masks** the
consequence. The unwritten-elements behavior itself remains unexplained and
is a correctness hazard: any triton kernel on this stack whose outputs are
allocated with `empty_like` and not fully written on first launches can
silently leak garbage. We're filing this issue to ask for investigation of
the root cause (kernel codegen in triton-ascend, launch/synchronization
path, or hardware quirk).

minimal reproduction script:
```
import os

os.environ["VLLM_WORKER_MULTIPROC_METHOD"] = "spawn"
os.environ["VLLM_ASCEND_ENABLE_NZ"] = "0"

from vllm import LLM, SamplingParams

PROMPT = "How are you?"

sampling_params = SamplingParams(temperature=0, max_tokens=100, seed=42)


def main() -> None:
    llm = LLM(
        "/mnt/data/models/Qwen3.6-27B-w8a8/",
        trust_remote_code=True,
        language_model_only=True,
        tensor_parallel_size=2,
        compilation_config={"cudagraph_mode": "FULL_DECODE_ONLY"},
    )

    out1 = llm.generate(PROMPT, sampling_params)[0].outputs[0]
    print("run 1 text:", repr(out1.text))
    print("run 1 token_ids:", out1.token_ids)
    print("run 1 finish_reason:", out1.finish_reason)

    out2 = llm.generate(PROMPT, sampling_params)[0].outputs[0]
    print("run 2 text:", repr(out2.text))
    print("run 2 token_ids:", out2.token_ids)
    print("run 2 finish_reason:", out2.finish_reason)


if __name__ == "__main__":
    main()
```

pr: https://github.com/vllm-project/vllm-ascend/pull/17003

**Comment by kunpengW-code** [comment]
The related fix PR has been linked. Thank you for the contribution!

**Comment by Absoler** [comment]
update pr : https://github.com/vllm-project/vllm-ascend/pull/17437

The main branch doesn't have this bug. And the second latest vllm-ascend image I could find is v0.26.0rc1, so I tested it and confirm the bug and the fix patch.
